docs/features/asc1/stateful/index.md

```diff
@@ -37,14 +37,24 @@ The `ClearStateProgram` handles the `ClearState` transaction and the `ApprovalPr
 
 The `goal` calls shown above in the orange boxes represent all the specific calls that can be made against a stateful smart contract and are described later in this document. These calls are also available in the SDKs. The teal-colored boxes represent the two required TEAL programs, the blue boxes are the state variables (local and global), and the yellow boxes represent the TEAL opcodes used to modify state. Modifying state is detailed in the next section.
 
+# Stateful Contract Arrays
+A set of arrays can be passed with any application transaction, which instructs the protocol to load additional data for use in the contract. These arrays are the *applications* array, the *accounts* array, the *assets* array, and the *arguments* array. The application array is used to pass other stateful smart contract IDs that can be used to read state for the specific contracts. The accounts array allows additional accounts to be passed to the contract for balance information and local storage. The assets array is used to pass a list of asset IDs that can be used to retrieve configuration and asset balance information. 
+
+The arguments array is used to pass standard arguments to the contract. The arguments array is limited to 16 arguments with a size limit of 2k for the total of arguments. See [Passing Arguments To Stateful Smart Contracts](#passing-arguments-to-stateful-smart-contracts) for more details on arguments.
+
+The other three arrays are limited to 8 total values combined, where the accounts array can have no more than four values. The values passed within these arrays can change per Application Transaction. Many opcodes that make use of these arrays take an integer parameter as an index into these arrays. The accounts and applications arrays contain the transaction sender and current application ID in the 0th position of the respective array. This shifts the contents of these two arrays by one slot. Most of the opcodes that use an index into these arrays also allow passing the actual value. For example, an address can be specified for an opcode that uses the accounts array, IDs can be specified for contracts and assets for opcodes that use the applications and assets arrays. These opcodes will fail if the specified value does not exist in the corresponding array. The use of each of these arrays is detailed throughout this guide.
```

Comment:
Suggestions:
"The other three arrays are limited to 8 total values combined, and of those, the accounts array can have no more than four values."

"For example, an address can be specified for an opcode that uses the accounts array. IDs can be specified for contracts and assets for an opcode that uses the applications or assets arrays, respectively."